product_template_default_weight/tests/test_weight_stock_picking.py

```diff
@@ -0,0 +1,127 @@
+# Copyright 2024 Akretion France (http://www.akretion.com/)
+# @author: Mathieu Delva <mathieu.delva@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from datetime import datetime
+
+from odoo.tests import tagged
+from odoo.tests.common import TransactionCase
+
+
+@tagged("post_install", "-at_install")
+class TestWeightStockPicking(TransactionCase):
+    def setUp(self):
+        super().setUp()
+        self.warehouse = self.env.ref("stock.warehouse0")
+        self.product_temp_id = self.env.ref(
+            "product.product_product_11_product_template"
+        )
+        self.product_id1 = self.env.ref("product.product_product_11")
+        self.product_id1.write(
+            {
+                "weight": 0,
+            }
+        )
+        self.product_id2 = self.env.ref("product.product_product_11b")
+        self.product_id2.write(
+            {
+                "weight": 100,
+            }
+        )
+
+        self.partner_id = self.env.ref("base.res_partner_12")
+
+    def test_weight_purchase_order_picking(self):
+        self.product_temp_id.write(
+            {
+                "weight": 50,
+            }
+        )
+        self.purchase_order_id = self.env["purchase.order"].create(
+            {
+                "partner_id": self.partner_id.id,
+                "date_planned": datetime.today(),
+                "order_line": [
+                    (
+                        0,
+                        0,
+                        {
+                            "product_id": self.product_id1.id,
+                            "product_qty": 2,
+                            "price_unit": 10,
+                        },
+                    ),
+                    (
+                        0,
+                        0,
+                        {
+                            "product_id": self.product_id2.id,
+                            "product_qty": 2,
+                            "price_unit": 10,
+                        },
+                    ),
+                ],
+            }
+        )
+
+        self.purchase_order_id.button_confirm()
+        picking_id = self.purchase_order_id.picking_ids
```

Comment:
This module doesn't depends on purchase...

Should we add the dependency in __manifest__.py ?